### PR TITLE
Socket pool metrics and refactorings

### PR DESCRIPTION
**Pull request by TonySkorik**
Add socket pool metrics;
Add optional node liveness check using socket pool;
Fix NRE when socket can't be created;
Minor refactorings;

### CODE REVIEW

File: src/Aer.Memcached.Client/CommandExecutor.cs [hunk 1]
```diff
@@ -202,11 +208,11 @@ public async Task DestroyAvailableSockets(int numberOfSocketsToDestroy, Cancella
     {
         try
         {
-            using var socket = await GetSocketAsync(node, token);
+            using var socket = await GetSocketAsync(node, isAuthenticateSocketIfRequired: true, token);
```
Comment (abarbolin) [inline src/Aer.Memcached.Client/CommandExecutor.cs:211]: Please check when memcached requires auth and health checker will try to check it without auth. Will it work?

File: src/Aer.Memcached.Client/Config/MemcachedConfiguration.cs [hunk 2]
```diff
@@ -56,14 +57,24 @@ public bool IsConfigured()
     public class MemcachedDiagnosticsSettings
```
Comment (abarbolin) [inline src/Aer.Memcached.Client/Config/MemcachedConfiguration.cs:57]: Please add all new configuration details in README

File: src/Aer.Memcached.Client/CommandExecutor.cs (hunk 1 above)
Comment (TonySkorik) [reply src/Aer.Memcached.Client/CommandExecutor.cs:211]: I tried to mimic the original behavior where the socket was created manually and did not go through the auth process.

Comment (abarbolin) [reply src/Aer.Memcached.Client/CommandExecutor.cs:211]: I guess it worked well then :)

**Pull request merged**